### PR TITLE
Add the nginx receiver

### PR DESCRIPTION
**Pull request by pjanotti**
Adding the ngnix receiver from contrib.

### CODE REVIEW

File: internal/components/components_test.go [hunk 1]
```diff
@@ -68,10 +68,11 @@ func TestDefaultComponents(t *testing.T) {
 		"kafka",
 		"kafkametrics",
 		"kubeletstats",
+		"lightprometheus",
```
[the remark this answers is not in the document]
Comment (pjanotti) [reply internal/components/components_test.go:71]: yes, it was the only one out of alphabetical order.

[the remark this answers is not in the document]
Comment (pjanotti) [reply internal/components/components_test.go:71]: or at least the only one that jumped out to my eyes

File: CHANGELOG.md [hunk 2]
```diff
@@ -163,7 +167,7 @@ This Splunk OpenTelemetry Collector release includes changes from the [opentelem
...
 - (Contrib) `pkg/stanza`: Switch to faster json parser lib for container operator ([#33929](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33929))
 - (Contrib) `telemetrygen`: telemetrygen `--rate` flag changed from Int64 to Float64 ([#33984](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33984))
-- (Contrib) `windowsperfcountersreceiver`: `windowsperfcountersreceiver` now appends an index number to additional instance names that share a name. An example of this is when scraping ` rocess(*)` counters with multiple running instances of the same executable. ([#32319](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32319))
+- (Contrib) `windowsperfcountersreceiver`: `windowsperfcountersreceiver` now appends an index number to additional instance names that share a name. An example of this is when scraping `rocess(*)` counters with multiple running instances of the same executable. ([#32319](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32319))
```
Comment (crobert-1) [inline CHANGELOG.md:170]:
```suggestion
- (Contrib) `windowsperfcountersreceiver`: `windowsperfcountersreceiver` now appends an index number to additional instance names that share a name. An example of this is when scraping `process(*)` counters with multiple running instances of the same executable. ([#32319](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32319))
```
Maybe this is what was meant?

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### 🚀 New components 🚀
+
+- (Splunk) Add `ngnix` receiver ([TBD](https://github.com/signalfx/splunk-otel-collector/pull/TBD/))
```
Comment (crobert-1) [inline CHANGELOG.md:7]:
```suggestion
- (Splunk) Add `ngnix` receiver ([5229](https://github.com/signalfx/splunk-otel-collector/pull/5229))
```

**Review by crobert-1 — APPROVED** [review]

**Comment by crobert-1** [comment]
Failing action output:
```
One or more Go files are not formatted correctly. Run 'make fmt' and push the changes.
```
and 
```
Generated code is out of date. Run 'make generate-metrics' and push the changes.
```

File: CHANGELOG.md (hunk 3 above)
Comment (pjanotti) [reply CHANGELOG.md:7]: Thank you :)

File: CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (pjanotti) [reply CHANGELOG.md]: This was supposed to fix a typo but I bungled it!

**Comment by pjanotti** [comment]
`make generate-metrics` is not changing anything on the mac, I will look more carefully if doesn't pass CI.

**Comment by pjanotti** [comment]
interesting that using the same docker it didn't need credentials on my box, but, needs on CI 🤔 anyway, I will add the credentials later.

**Pull request merged**